### PR TITLE
Suppress country selection per document type

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
For integrators who only accept a document from a specific country or already have the country data for the user's document, they do not want users to select a country after selecting a document type. The current `showCountrySelection` option for the Document step is also limited to only take effect for a single preselected non-passport document. There is no simple way to skip the country selection screen for a specific non-passport document type when there are multiple document types enabled.

# Solution
Add support for individual document type country configuration For backward compatibility, this works as an alternative to the current boolean value for initialising the SDK. 

Full Document step config structure is as follows:
```
{
    type: 'document', 
    options: {
        documentTypes: {
            passport: <Boolean>,
            driving_licence: <Boolean> | <Object: { country: <3-letter ISO 3166-1 alpha-3 code string | null> }>,
            residence_permit: <Boolean> | <Object: { country: <3-letter ISO 3166-1 alpha-3 code string | null> }>,
            national_identity_card: <Boolean> | <Object: { country: <3-letter ISO 3166-1 alpha-3 code string | null> }>
        },
       showCountrySelection: <Boolean>
    }
}
```

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [x] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Review by ethan605 — APPROVED** [review]
LGTM

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

File: README.md [hunk 1]
```diff
@@ -429,6 +429,58 @@ A number of options are available to allow you to customise the SDK:
...
+              "passport": true,
+              "national_identity_card": true
+            }
+          }
+        },
+        "complete"
+      ]
+    }
+    ```
+
```
Comment (stefaniacardenas) [inline README.md:483]: This behaviour is much easier to understand

**Review by stefaniacardenas — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

File: README.md (hunk 1 above)
Comment (kopijunkie) [reply README.md]: Still have the `showCountrySelection` option, but we'll deprecate it at some point since it's rather confusing. 😅

Comment (ethan605) [reply README.md]: @kopijunkie if we plan to deprecate it soon, can we add a note in README to indicate that?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

File: README.md [hunk 2]
```diff
@@ -429,6 +429,58 @@ A number of options are available to allow you to customise the SDK:
 
     The list of document types visible to the user can be filtered by using the `documentTypes` option. The default value for each document type is `true`. If `documentTypes` only includes one document type, users will not see both the document selection screen and country selection screen and instead will be taken to the capture screen directly.
```
Comment (seewah) [inline README.md:430]: TRIVIAL: i know it is not part of this PR, but "users will not see either the document selection screen or the country selection screen" reads better :)

**Review by seewah — CHANGES_REQUESTED** [review]
i see that issuing_country value is not sent when it is being preset - i test with ESP. The specified issuing_country value should be sent in this case. Thanks

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Review by ethan605 — APPROVED** [review]
LGTM!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1232-pr-onfido-sdk-ui-onfido.surge.sh

**Review by seewah — APPROVED** [review]

**Pull request merged**